rest/admin_api.go

```diff
@@ -452,11 +454,21 @@ func (h *handler) handleSGCollect() error {
 }
 
 // streamPipe streams output from pipeReader into the response.
-func streamPipe(resp http.ResponseWriter, pipeReader *io.PipeReader) {
+func streamPipe(ctx context.Context, resp http.ResponseWriter, pipeReader *io.PipeReader) {
 	buffer := make([]byte, 1024) // 1kB buffer
 	for {
+		// Handle cancellation
+		select {
+		case <-ctx.Done():
+			return
```

Comment:
Should it be calling `pipeReader.Close()` here?  (or moved to a defer statement?)